crates/dekoder/src/inventory.rs

```diff
@@ -0,0 +1,87 @@
+//! Assets manager.
+use glob::glob;
+use hashbrown::HashMap;
+use std::ffi::OsString;
+use std::fs::read_to_string;
+use std::path::PathBuf;
+use yaml_rust2::{Yaml, YamlLoader};
+
+use crate::{EKOError, Result};
+
+/// Headers are in yaml files.
+const HEADER_EXT: &'static str = "*.yaml";
+
+/// Header type in an inventory.
+pub(crate) trait HeaderT {
+    /// Load from yaml.
+    fn load_from_yaml(yml: &Yaml) -> Result<Self>
+    where
+        Self: Sized;
+    /// Comparator.
+    fn eq(&self, other: &Self, ulps: i64) -> bool;
+}
+
+/// Value type in an inventory.
+pub(crate) trait ValueT {
```

Comment:
Similar to `HeaderT` you could replace `ValueT` with `From<&Path>` or even more generally `From<Read>`.

Reply:
should I then hardcode the file suffix `npz.lz4`? for our scope here is fine I think, but this would basically lift everything into `Inventory`; Pushing this idea, it would make our "values" (in the `Inventory`) always of what is currently called `Operator`, again I think this is fine here as they will always be 4D arrays (possibly with errors). It is only the "keys" which need to be of different types. Do you agree with this plan? (lift everything)

Reply:
Yes!

Reply:
done in 1a696a18